### PR TITLE
VFB-227 Resolve Supabase security advisory errors

### LINKED ISSUE #272: See date/time when a User last logged into their account
### Is your feature request related to a problem? If so, please describe.

_No response_

### Describe the solution / feature you'd like

On the > Admin Page > Users Table
Add a column which shows a date and time stamp for when each user last logged in.

This would be helpful for identifying in-active accounts.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by stonelink**
There were two types of security advisory errors:
- views had the the default SECURITY_DEFINER state, meaning that the view had Postgres-level permissions even if the invoking user doesn't have them
- contents of the auth table were being exposed in the `users_plus` view

Because the fix for the `users_plus` issue was to add more info to the `profiles` table, I also added each user's Last Sign In Date, which resolves #272 

## Checklist
- [X] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA
- [X] I have self-reviewed this PR
- [X] Make sure you've verified it works via `npm run dev`
- [X] Make sure you've verified it works via `npm run build` and `npm run start`
- [X] Make sure you've fixed all linting problems with `npm run lint_fix`
- [X] Make sure you've tested via `npm run test:e2e`

If you have made any changes to the database...
  - [X] The migration files are up-to-date with my final set up (`npx supabase db diff -f <name_of_migration>` should create nothing at this point)
  - [X] I have updated the typescript definitions for the database with `npm run db:local:generate_types`
  - [-] I have modified the seed in `supabase/seed/seed.mts` if appropriate
  - [-] If I have modified the seed, I have also generated the seed with `npm run db:generate_seed` 
  - [X] With my final set up, I can run `npm run dev:reset_supabase` without any errors.
  - [x] (Just before merging the ticket) I have updated the timestamps of the new migration files so that they are after all existing migration files.
  - [-] If I have added constraints e.g. `set not null`, I have taken reasonable measures to ensure constraint violations do not happen when the migration occurs. 
  - [x] I have checked that migration can happen successfully without resetting the database, doing the following.
    - Make sure you have rebased your branch onto dev or merged dev into your branch.
    - Checkout the dev branch
    - Run `npm run post_checkout` to reset the database, including the seed data.
    - Checkout your branch
    - Run `npx supabase migrations up --include-all --local` to run all outstanding migration files
    - Check that the resulting database is what you expect, bar any seed data changes.

### CODE REVIEW

File: supabase/tools/createUserForEachRole.ts [hunk 1]
```diff
@@ -69,12 +69,15 @@ async function createUser(supabase: SupabaseClient, userProfile: userProfile): P
     }
 
     if (data) {
+        const nowTimestamp = new Date().toISOString();
```
Comment (Stuart-Knowles-SW) [inline supabase/tools/createUserForEachRole.ts]: Could we use a constant time for this? I doubt it'll matter much in practice, but I have a preference for seed data (which this sort of is) to be static. Not too fussed though

**Review by Stuart-Knowles-SW — APPROVED** [review]
One tiny comment but nothing too important. Overall looks great!

File: supabase/tools/createUserForEachRole.ts (hunk 1 above)
Comment (stonelink) [reply supabase/tools/createUserForEachRole.ts]: Good point, fixed.

**Comment by stonelink** [comment]
## QA steps
Login as different roles (admin, volunteer) in turn and check that all fields in in tables and modals that are populated from the following are working as expected:
[ ] parcels - including events/status
[ ] clients - including family details
[ ] reports

Admins only:
[ ] users - including checking 'last sign in' time
[ ] audit log

**Pull request merged**